### ISSUE TITLE
nsi.guide - future plans

### ISSUE THREAD

**Issue by UKChris-osm** [osmlab/name-suggestion-index#3990]
Over on the now closed PR #3770 a couple of issues have come up, and decision has also moved on to the future of the web site, so I thought rather than keeping that discussion going in a closed PR, I would open up a new issue.

Some of the ideas for nsi.guide include:

- Extra consideration for colour blindness when it comes to the current red, yellow, green markers.
- Using locations to limit the Overpass Turbo query to just Countries in which the brand operates.
- Distinguishing between similar / same brands that are of a different key / value type.
- Replacing the links to Overpass Turbo with an in-house NSI map that uses Overpass directly and filters things via JavaScript for a better user experience with more NSI-centric information bubbles and legends.

**Comment by UKChris-osm** [comment]
One thing I am curious about, is the web site hosting for [nsi.guide](https://nsi.guide)

Currently I think it's just on GitHub Pages, which doesn't provide any form of server side abilities (scripting, SQL, etc) but is there any possibility of using the same hosting that OSM uses (for example, https://nsi.openstreetmap.org/) to allow for a more dynamic, more ingirted web site?

**Comment by bhousel** [comment]
Yes it's just GitHub Pages for now, and I think that works ok - I can't think of anything that we would need a server for.  I'd prefer to keep this project distinct from OpenStreetMap for now.

**Comment by UKChris-osm** [comment]
> I can't think of anything that we would need a server for.

I was thinking of things like enabling a user to submit a brand idea, with Wikidata / Wikipedia values, without them having to know how to use GitHub or requiring a GitHub account, such as using an HTML Form to submit the data, and have the server side add that data to a file (submissions.json or something) so that the NSI project could process those suggestions in batches.

**Comment by Adamant36** [comment]
> was thinking of things like enabling a user to submit a brand idea, with Wikidata / Wikipedia values, without them having to know how to use GitHub or requiring a GitHub account, such as using an HTML Form to submit the data, and have the server side add that data to a file (submissions.json or something) so that the NSI project could process those suggestions in batches.

Maybe it could be done on iD Editors side. Like a form in the editor that would submit it to nsi.guide for one of us to deal with and create an entry for. I don't know how many times I've found a new brand while editing, made a note of it in my note taking app, and then never created an issue or PR for it.  So something like that would be really useful IMO. Especially if the entry could be auto-generated in the guide and not have to go through Github. I don't see why something like that couldn't be done for at least the basic details. Then we could add the Wikidata/Wikipedia information as needed, but it would save us some work and increse contributations. Especially from non-English speakers. Github is great for some things, but its a barrier for a lot of people. So it would be great if we didn't have to deal with issues, PRs, etc at least for basic trivial things. Like "here's a brand that hasn't been added yet."

**Comment by michaelblyons** [comment]
> - Using locations to limit the Overpass Turbo query to just Countries in which the brand operates.
> - Replacing the links to Overpass Turbo with an in-house NSI map that uses Overpass directly and filters things via JavaScript for a better user experience with more NSI-centric information bubbles and legends.

I would also like to see (either on OT or a homebrew map) the `matchNames` matches in yet another color. After a dedup, you're left with NSI.guide saying "60 items match," but the OT link only shows you the "winner" of the deduplication.
